### PR TITLE
doc conda-gcc-specs and -sysroot in user faq

### PR DESCRIPTION
**Pull request by timsnyder**
capture takeaway of conda-forge/ctng-compilers-feedstock#66

also document conda-gcc-specs (conda-forge/ctng-compilers-feedstock#91) and
symlinks (conda-forge/ctng-compiler-activation-feedstock#51)

PR Checklist:

- [X] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [X] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [X] put any other relevant information below

### CODE REVIEW

File: src/user/faq.rst [hunk 1]
```diff
@@ -73,6 +73,56 @@ FAQ
     
   are a telltale sign that you are lacking compilers.
 
+.. _faq_compiler_required_options:
+
+:ref:`(Q) <faq_compiler_required_options>` **Why don't the C/C++ compilers automatically know how to find libraries installed by conda?**
+
+  All of our toolchains are built as cross-compilers (even when they are built to run on the same
+  architecture that they are targeting).  We do this because it makes it possible to then install
+  them anywhere like any other conda package.  As a result, the buildin search path for the
```
Comment (beckermr) [inline src/user/faq.rst:82]: builtin

File: src/user/faq.rst [hunk 2]
```diff
@@ -73,6 +73,56 @@ FAQ
...
+  ``CMAKE_ARGS`` variable with many arguments the conda-forge community finds helpful for
+  configuring cmake build flows.  Of particular note, the activation scripts add the
+  ``CONDA_PREFIX/include`` and ``CONDA_PREFIX/lib`` paths to the appropriate ``FLAGS`` environment
+  variables (``CLAGS``, ``CPPFLAGS``, ``LDFLAGS``, etc) so that many build systems will pick them up correctly.
+
+  What do you do if you have custom ``FLAGS`` that your project requires for it's build or you can't
+  build with some of the flags supplied by conda-forge?  What if you are building something that
+  is setup for cross-compiling and expects ``CC`` to contain the name of the target toolchain but
+  wants to be able to build some things for the build-host to use during the build by just calling
+  ``gcc``?  You can filter out the
```
Comment (beckermr) [inline src/user/faq.rst:101]: Is this missing text?

**Review by beckermr — APPROVED** [review]

**Review by beckermr — COMMENTED** [review]
Meant to leave comments.

File: src/user/faq.rst (hunk 1 above)
Comment (timsnyder) [reply src/user/faq.rst:82]:
```suggestion
  them anywhere like any other conda package.  As a result, the builtin search path for the
```

File: src/user/faq.rst (hunk 2 above)
Comment (timsnyder) [reply src/user/faq.rst:101]:
```suggestion
  ``gcc``? 
```

Comment (timsnyder) [reply src/user/faq.rst:101]: Yeah, I was starting to type something and decided I was getting long-winded and didn't remove it.

**Comment by beckermr** [comment]
Thank you!

**Pull request merged**